product_other_refereces/models/product_product.py

```diff
@@ -0,0 +1,25 @@
+# © 2023 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import models, api
+from odoo.osv import expression
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    @api.model
+    def _name_search(self, name, args, operator="ilike", limit=None, name_get_uid=None):
+        res = super(ProductProduct, self)._name_search(
+            name, args=args, operator=operator, limit=limit, name_get_uid=name_get_uid
+        )
+        res_ids = list(res)
```

Comment:
is actually returned `res` as a list? In that case, this line could be rewritten or skipped

Reply:
Sometimes it returns a list and sometimes an odoo object: odoo.osv.query.Query and for those cases I convert it to a list